src/test/java/io/supertokens/test/passwordless/api/PasswordlessCreateCodeAPITest2_10.java

```diff
@@ -0,0 +1,458 @@
+        resendCodeRequestBody.addProperty("deviceId", createResp.get("deviceId").getAsString());
+        resendCodeRequestBody.addProperty("userInputCode", exampleCode);
+
+        JsonObject resendResponse = HttpRequestForTesting.sendJsonPOSTRequest(process.getProcess(), "",
+                "http://localhost:3567/recipe/signinup/code", resendCodeRequestBody, 1000, 1000, null,
+                Utils.getCdiVersion2_10ForTests(), "passwordless");
+
+        assertEquals("USER_INPUT_CODE_ALREADY_USED_ERROR", resendResponse.get("status").getAsString());
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    @Test
+    public void testResendWithNonExistantDeviceId() throws Exception {
+        String[] args = { "../" };
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        if (StorageLayer.getStorage(process.getProcess()).getType() != STORAGE_TYPE.SQL) {
+            return;
+        }
+
+        JsonObject resendCodeRequestBody = new JsonObject();
+        resendCodeRequestBody.addProperty("deviceId", "JWlE/V+Uz8qgaTyFkzOI4FfRrU6fBH85ve2GunoPpz0=");
+
+        JsonObject resendResponse = HttpRequestForTesting.sendJsonPOSTRequest(process.getProcess(), "",
+                "http://localhost:3567/recipe/signinup/code", resendCodeRequestBody, 1000, 1000, null,
+                Utils.getCdiVersion2_10ForTests(), "passwordless");
+
+        assertEquals("RESTART_FLOW_ERROR", resendResponse.get("status").getAsString());
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    @Test
+    public void testAllIds() throws Exception {
+        String[] args = { "../" };
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        if (StorageLayer.getStorage(process.getProcess()).getType() != STORAGE_TYPE.SQL) {
+            return;
+        }
+
+        JsonObject createCodeRequestBody = new JsonObject();
+        createCodeRequestBody.addProperty("email", "test@example.com");
+        createCodeRequestBody.addProperty("phoneNumber", "+442071838750");
+        createCodeRequestBody.addProperty("deviceId", "+442071838750");
+
+        HttpResponseException error = null;
+        try {
+            HttpRequestForTesting.sendJsonPOSTRequest(process.getProcess(), "",
+                    "http://localhost:3567/recipe/signinup/code", createCodeRequestBody, 1000, 1000, null,
+                    Utils.getCdiVersion2_10ForTests(), "passwordless");
+        } catch (HttpResponseException ex) {
+            error = ex;
+        }
+
+        assertNotNull(error);
+        assertEquals(400, error.statusCode);
```

Comment:
You should also check the actual message returned (in this test and the ones below this).